File: frontend/src/api/reminders.ts
import client from './client'
import type { Reminder } from '../types'

export const getReminders = (post_id?: number) =>
  client.get<Reminder[]>('/reminders', { params: post_id != null ? { post_id } : {} }).then(r => r.data)

export const createReminder = (data: { post_id: number; message: string; send_at: string | null }) =>
  client.post<Reminder>('/reminders', data).then(r => r.data)

export const deleteReminder = (id: number) =>
  client.delete(`/reminders/${id}`).then(r => r.data)

File: backend/routers/reminders.py
from typing import List, Optional
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from database import get_db
from models.reminder import Reminder
from schemas.reminder import ReminderCreate, ReminderRead

router = APIRouter(prefix="/api/reminders", tags=["reminders"])


@router.get("", response_model=List[ReminderRead])
def list_reminders(post_id: Optional[int] = None, db: Session = Depends(get_db)):
    q = db.query(Reminder)
    if post_id is not None:
        q = q.filter(Reminder.post_id == post_id)
    return q.order_by(Reminder.send_at).all()


@router.post("", response_model=ReminderRead, status_code=201)
def create_reminder(data: ReminderCreate, db: Session = Depends(get_db)):
    reminder = Reminder(post_id=data.post_id, message=data.message, send_at=data.send_at)
    db.add(reminder)
    db.commit()
    db.refresh(reminder)
    return reminder


@router.delete("/{reminder_id}")
def delete_reminder(reminder_id: int, db: Session = Depends(get_db)):
    r = db.get(Reminder, reminder_id)
    if not r:
        raise HTTPException(404, "Reminder not found")
    db.delete(r)
    db.commit()
    return {"ok": True}
